zsh/functions/envup

```diff
@@ -2,7 +2,7 @@
 
 function envup() {
   if [ -f .env ]; then
-    export $(cat .env)
+    export $(sed '/^#/ d' .env)
```

Comment:
`sed` certainly works here, but it feels a bit odd to me. My lean would be to use `grep` as I associate that with finding / filtering lines, and `sed` with transforming, but that is sort of an arbitrary and perhaps artificial distinction.

If you did want to use `grep`, this could be:

```
export $(grep -v '^#' .env)
```

Reply:
I'm cool with using sed like that (I guess because I write lines like that in vi?). But how about a more permissive regexp? Is it unreasonable to have e.g. `/^ *#/` -- allow whitespace before the comment?